packages/patternfly-3/patternfly-react-extensions/src/components/CatalogTile/CatalogTile.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { OverlayTrigger, Tooltip } from 'patternfly-react';
+
+import CatalogTileBadge from './CatalogTileBadge';
+
+const CatalogTile = ({
+  id,
+  className,
+  featured,
+  iconImg,
+  badges,
+  title,
+  vendor,
+  description,
+  maxDescriptionLength,
+  truncateDescriptionFn,
+  ...props
+}) => {
+  const classes = classNames('catalog-tile-pf', { featured }, className);
+
+  const defaultTruncateDescription = (text, max) => {
+    if (typeof text === 'string' && text.length <= max) {
```

Comment:
for this truncation / tooltip functionality, we had been using the Foreman [react-ellipsis-with-tooltip](https://github.com/amirfefer/react-ellipsis-with-tooltip) package.

example usage:
https://github.com/ManageIQ/manageiq-v2v/blob/505e87ec24e191340154bb2d83718b9d2843b728/app/javascript/react/screens/App/Overview/screens/MappingWizard/components/DualPaneMapper/DualPaneMapperListItem.js#L36

I like the option to override the function like you've provided though, but maybe we should standardize the default function a bit since it so common. Could we compare the react-ellipsis-with-tooltip function with this one and maybe throw it in an exposed helper? 

Not pressing, but just a consideration/mention.